authentication/serializers.py

```diff
@@ -0,0 +1,106 @@
+        re_password = attrs.pop("re_password")
+        password = attrs.get("password")
+        try:
+            validate_password_long(password)
+            validate_password_include_symbols(password)
+        except ValidationError as error:
+            print(error)
+            raise serializers.ValidationError({"password": error.message})
+        if attrs["password"] == re_password:
+            return attrs
+        else:
+            raise serializers.ValidationError(
+                {"password": "Passwords don't match."})
+
+    def create(self, validated_data: dict):
+        if User.objects.filter(person_email=validated_data["person_email"]):
+            raise serializers.ValidationError(
+                {"error": "Email is already registered"})
+        comp_registered = validated_data.get("comp_registered")
+        comp_is_startup = validated_data.get("comp_is_startup")
+        if not comp_registered and not comp_is_startup:
+            raise serializers.ValidationError({"error":
+                                               "Please choose who you represent"})
+        if comp_registered and comp_is_startup:
+            raise serializers.ValidationError({"error":
+                                               "Please choose either registered or startup"})
+        user = User.objects.create(**validated_data)
+        user.set_password(validated_data["password"])
+        user.is_active = True
+        user.save()
+        return user
+
+
+class UserListSerializer(UserSerializer):
+
+    class Meta(UserSerializer.Meta):
+        model = User
+        fields = (
+            "person_email",
+            "person_name",
+            "person_surname",
+            "comp_name",
+            "comp_registered",
+            "comp_is_startup",
+        )
+
+
+class UserTokenCreateSerializer(TokenCreateSerializer):
+    password = serializers.CharField(
+        required=False, style={"input_type": "password"})
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.user = None
+        self.fields[settings.DJOSER["LOGIN_FIELD"]] = serializers.CharField(
+            required=False)
+
+    def validate(self, attrs):
+        password = attrs.get("password")
+        params = {settings.DJOSER["LOGIN_FIELD"]: attrs.get(settings.DJOSER["LOGIN_FIELD"])}
+        self.user = authenticate(
+            request=self.context.get("request"), **params, password=password
+        )
+        if not self.user:
+            self.user = User.objects.filter(**params).first()
+            if self.user and not self.user.check_password(password):
+                raise serializers.ValidationError({"error":
+                                                   "Email or password is incorrect"})
+        if self.user and self.user.is_active:
+            return attrs
+        raise serializers.ValidationError({"error":
+                                           "Email or password is incorrect"})
```

Comment:
You can also change it to if-elif-else here.

Reply:
It is the logic of Djoser TokenCreateSerializer. Maybe it's no need to change it here.